dashboard/app/models/pd/regional_partner_contact.rb

```diff
@@ -70,4 +72,28 @@ def validate_district_fields
       add_key_error(:school_district_data) unless hash[:school_name].presence && hash[:school_zipcode]
     end
   end
+
+  def update_regional_partner
+    return if sanitize_form_data_hash[:grade_levels] == ['Elementary School']
+
+    school_district = SchoolDistrict.find_by_id(sanitize_form_data_hash[:school_district])
+
+    return unless school_district
+
+    possible_partners = school_district.regional_partners_school_districts
+
+    if possible_partners.size == 1
+      self.regional_partner = possible_partners.first.regional_partner
+    elsif possible_partners.size > 1
+      grade_levels = sanitize_form_data_hash[:grade_levels]
+
+      if grade_levels.include? 'High School'
+        self.regional_partner = possible_partners.find_by(course: 'csp').regional_partner
+      elsif grade_levels.include? 'Middle School'
```

Comment:
should these be in constants? Where are the values provided for the form?

Reply:
They come from regional_partneres_school_districts. I can make them a constant